### PR TITLE
Build ESM for packages

### PR DESCRIPTION
**Pull request by dr3**
Contributes to https://github.com/bbc/psammead/issues/557

**Overall change:** Builds ESM for every package (not used yet)

**Code changes:**

- Builds ESM for every package.
- Updates our babel env config to match simorgh.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -1,4 +1,27 @@
-const presets = ['@babel/preset-env', '@babel/preset-react'];
+const presets = [
+  '@babel/preset-react',
+  [
+    '@babel/preset-env',
+    {
+      targets: {
```
Comment (jamesbhobbs) [inline babel.config.js:6]: this doesn't match this repo's readme, can you correct the readme pls

**Review by chris-hinds — APPROVED** [review]

**Review by jroebu14 — APPROVED** [review]
Nice one 👍

**Comment by chris-hinds** [comment]
I have checked this locally by running `npm run build` the esm and dist structure looks good to me.

**Comment by dr3** [comment]
Tested by Chris due to intricacies with babel, should have no regression, will fix forward if does.

**Pull request merged**